src/input/keyboard/mod.rs

```diff
@@ -697,6 +786,12 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
             };
         }
 
+        if leds_changed {
+            let led_state = internal.led_state;
```

Comment:
```suggestion
            let led_state = internal.led_state;
            std::mem::drop(internal);
```
Input deadlocks my beloved

Reply:
`internal` is a `&mut` in this function, I instead made sure the scope of `internal` ends before `data.led_state_changed` is invoked.